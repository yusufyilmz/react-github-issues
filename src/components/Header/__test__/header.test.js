

import React from 'react';
import {Header} from '../index';
import { shallow } from 'enzyme'
import { Menu } from '../../Menu'
import { Logo } from '../../Logo'
import {  Row, Col } from 'react-bootstrap';


describe('HEADER COMPONENT', () => {

    let message, wrapper;

    beforeEach(() => {
        message = 'test';
        wrapper = shallow(
            <Header message={message} />
        );
    });
   
      
    test('renders element correctly', () => {
        expect(wrapper.find('ContainerWrapper').length).toBe(1);
    });

    test('renders element correctly', () => {
        expect(wrapper.find(Menu).length).toBe(1);
    });

    test('renders element correctly', () => {
        expect(wrapper.find(Logo).length).toBe(1);
    });


    test('renders text correctly', () => {
        expect(wrapper.find(Col).at(0).prop('md')).toBe(4);
    });

    test('snapshot correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
})
