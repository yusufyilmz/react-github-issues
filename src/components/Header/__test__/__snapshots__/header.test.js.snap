// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HEADER COMPONENT snapshot correctly 1`] = `
<ContainerWrapper>
  <ForwardRef(Bootstrap(Row))>
    <ForwardRef(Bootstrap(Col))
      md={4}
      xs={4}
    >
      <Logo />
    </ForwardRef(Bootstrap(Col))>
    <ForwardRef(Bootstrap(Col))>
      <Menu />
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(Row))>
</ContainerWrapper>
`;
